utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java

```diff
@@ -275,16 +276,125 @@ private static DescriptorProto messageElementToDescriptorProto(
         ProtobufMessage message = new ProtobufMessage();
         message.protoBuilder().setName(messageElem.getType().getSimpleName());
 
+        Map<Integer, DescriptorProto> allNestedTypes = new TreeMap<>();
+        Map<Integer, FieldDescriptorProto> allFields = new LinkedHashMap<>();
+
         for (Type type : messageElem.getNestedTypes()) {
             if (type instanceof MessageType) {
-                message.protoBuilder().addNestedType(
-                    messageElementToDescriptorProto((MessageType) type, schema, element));
+                allNestedTypes.put(type.getLocation().getLine(),
```

Comment:
This might overwrite two messages that are on the same line. Good to store Location itself as the Key as it contains both line and column.